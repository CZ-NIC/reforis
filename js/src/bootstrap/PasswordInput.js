/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, {useState} from 'react';
import PropTypes from 'prop-types';

import Input from './Input';

PasswordInput.propTypes = {
    label: PropTypes.string.isRequired,
    error: PropTypes.string,
    value: PropTypes.string,
    helpText: PropTypes.string,
};

export default function PasswordInput(props) {
    const [isHidden, setHidden] = useState(true);

    const InputField = isHidden ? Input('password') : Input('text');
    return <InputField
        autoComplete={isHidden ? 'new-password' : null}
        {...props}
    >
        <div className="input-group-append">
            <button className="input-group-text" onClick={() => setHidden(isHidden => !isHidden)}>
                <i className={'fa ' + (isHidden ? 'fa-eye' : 'fa-eye-slash')}/>
            </button>
        </div>
    </InputField>
}

