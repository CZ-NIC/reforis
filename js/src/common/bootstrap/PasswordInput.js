/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React, {useState} from 'react';
import PropTypes from 'prop-types';

import Input from './Input';

PasswordInput.propTypes = {
    /** Field label. */
    label: PropTypes.string.isRequired,
    /** Error message. */
    error: PropTypes.string,
    /** Password value. */
    value: PropTypes.string,
    /** Help text message. */
    helpText: PropTypes.string,
    /** Use show/hide password button. */
    withEye: PropTypes.bool,
};

export default function PasswordInput({withEye, ...props}) {
    const [isHidden, setHidden] = useState(true);
    return <Input
        type={withEye ? isHidden ? 'password' : 'text' : 'password'}
        autoComplete={isHidden ? 'new-password' : null}
        {...props}
    >
        {withEye ?
            <div className="input-group-append">
                <button className="input-group-text" onClick={e => {
                    e.preventDefault();
                    setHidden(isHidden => !isHidden);
                }}>
                    <i className={'fa ' + (isHidden ? 'fa-eye' : 'fa-eye-slash')}/>
                </button>
            </div>
            : null}
    </Input>
}
