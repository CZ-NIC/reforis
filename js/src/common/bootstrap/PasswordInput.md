Password Bootstrap component input with label and predefined sizes and structure for using in foris forms.
Can be used with "eye" button, see example.

All additional `props` are passed to the `<input type="password">` HTML component.

```js
import {useState} from 'react';
const [value, setValue] = useState('secret');

<PasswordInput
    withEye
    value={value}
    label="Some password" 
    helpText="Read the small text!"
    onChange={target => setValue(target.value)}
/>
```
