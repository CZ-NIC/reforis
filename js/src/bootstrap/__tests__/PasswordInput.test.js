/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */
import renderer from 'react-test-renderer';
import PasswordInput from '../PasswordInput';
import React from 'react';

describe('<PasswordInput/>', () => {
    it('Render password input', () => {
        const tree = renderer
            .create(
                <PasswordInput
                    label="Test label"
                    checked
                    helpText="Some help text"
                    value="Some password"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});