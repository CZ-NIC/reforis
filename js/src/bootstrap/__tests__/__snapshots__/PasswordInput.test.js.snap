// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput/> Render password input 1`] = `
<div
  class="form-group row"
>
  <label
    class="form-control-label col-sm-3"
    for="1"
  >
    Test label
  </label>
  <div
    class="col-sm-8"
  >
    <div
      class="input-group"
    >
      <input
        autocomplete="new-password"
        class="form-control "
        id="1"
        type="password"
        value="Some password"
      />
      <div
        class="input-group-append"
      >
        <button
          class="input-group-text"
        >
          <i
            class="fa fa-eye"
          />
        </button>
      </div>
    </div>
    <div
      class="invalid-feedback"
    />
    <small
      class="form-text text-muted"
    >
      Some help text
    </small>
  </div>
</div>
`;
